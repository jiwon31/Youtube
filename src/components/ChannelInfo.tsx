import { useQuery } from "@tanstack/react-query";
import { useYoutubeApi } from "context/YoutubeApiContext";

type ChannelInfoProps = {
  id: string;
  name: string;
};

export default function ChannelInfo({ id, name }: ChannelInfoProps) {
  const { youtube } = useYoutubeApi();
  const { data: url } = useQuery(
    ["channel", id],
    async () => youtube.channelImageURL(id),
    { staleTime: 1000 * 60 * 5 }
  );

  return (
    <div className="flex items-center my-4 mb-8">
      {url && (
        <img className="w-10 h-10 rounded-full mr-2" src={url} alt={name} />
      )}
      <p className="text-lg font-medium">{name}</p>
    </div>
  );
}
