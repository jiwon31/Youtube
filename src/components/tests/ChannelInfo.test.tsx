import { render, screen, waitFor } from "@testing-library/react";
import ChannelInfo from "components/ChannelInfo";
import { Route } from "react-router-dom";
import { withAllContext, withRouter } from "tests/utils";

describe("ChannelInfo", () => {
  const fakeYoutube = {
    channelImageURL: jest.fn(),
  };
  afterEach(() => fakeYoutube.channelImageURL.mockReset());

  it("renders", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");
    const { asFragment } = renderChannelInfo();

    await waitFor(() => screen.getByRole("img"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without channel thumbnail", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => {
      throw new Error("error");
    });
    renderChannelInfo();

    expect(screen.queryByRole("img")).toBeNull();
  });

  it("renders with channel thumbnail", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");
    renderChannelInfo();

    await waitFor(() => expect(screen.getByRole("img")).toBeInTheDocument());
  });

  function renderChannelInfo() {
    return render(
      withAllContext(
        withRouter(
          <Route path="/" element={<ChannelInfo id="1" name="channel" />} />
        ),
        fakeYoutube
      )
    );
  }
});
