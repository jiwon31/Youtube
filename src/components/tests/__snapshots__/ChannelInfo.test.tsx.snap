// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders 1`] = `
<DocumentFragment>
  <div
    class="flex items-center my-4 mb-8"
  >
    <img
      alt="channel"
      class="w-10 h-10 rounded-full mr-2"
      src="url"
    />
    <p
      class="text-lg font-medium"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
